packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -242,35 +317,56 @@ const getInStyle = ({ x, width, y, height }: CompletedBarData) => ({
  * - [BarPlot API](https://mui.com/x/api/charts/bar-plot/)
  */
 function BarPlot(props: BarPlotProps) {
-  const completedData = useAggregatedData();
+  const { completedData, masksData } = useAggregatedData();
   const { skipAnimation, onItemClick, ...other } = props;
+  const transition = useTransition(
+    completedData.map((d) => ({
+      ...d,
+      mask: masksData[d.maskId],
+    })),
+    {
+      keys: (bar) => `${bar.seriesId}-${bar.dataIndex}`,
+      from: getOutStyle,
+      leave: getOutStyle,
+      enter: getInStyle,
+      update: getInStyle,
+      immediate: skipAnimation,
+    },
+  );
 
-  const transition = useTransition(completedData, {
-    keys: (bar) => `${bar.seriesId}-${bar.dataIndex}`,
-    from: getOutStyle,
-    leave: getOutStyle,
-    enter: getInStyle,
-    update: getInStyle,
-    immediate: skipAnimation,
-  });
   return (
     <React.Fragment>
-      {transition((style, { seriesId, dataIndex, color, highlightScope }) => (
-        <BarElement
-          id={seriesId}
-          dataIndex={dataIndex}
-          highlightScope={highlightScope}
-          color={color}
-          {...other}
-          onClick={
-            onItemClick &&
-            ((event) => {
-              onItemClick(event, { type: 'bar', seriesId, dataIndex });
-            })
-          }
-          style={style}
-        />
-      ))}
+      {transition((style, { seriesId, dataIndex, color, highlightScope, maskId }) => {
+        return (
+          <BarGroup
+            key={`${seriesId}-${dataIndex}`}
```

Comment:
A bit weird to have as many mask as bar. Would it make sens to define a `<BarGroupClipPath clipingId={maskId} style={ ... } />` that defines the mask at put an id (will need to add the chartId to avoid conflicts)

Render those compoennts in their dedicated loop. And keep our current BarElement with ```clipPath={`url(#${maskId})`}```


Bonus point for not rendering the `BarGroupClipPath` if borderRadius is undefined or is set to 0 :)

Reply:
My initial idea was to render all the masks, and only animate those. But it didn't seem visually correct, as the masks would "slide over" stacks, while currently stacks grow individually.

Another try was to have the statics masks, but allowing the stacks to grow. This solved the previous problem, but the stacks were "flat" (no border radius), while they were growing, which looked odd.

In the end I had to animate both the mask and the stack to have the behaviour I was looking for, but then I didn't know how to align the animations to play at exactly the same time, while adding/removing stacks also looked odd because each of the animations reacted differently.

My current solution simply directly animates them all in the same `useTransition`, with the sideeffect that we have as many masks as stacks, as you pointed out.

Reply:
I figured out I could simply use a new `useTransition` with the mask data 🙃, this way we can remove unneeded masks in when stacking